### ISSUE TITLE
Error when Build source path matches current directory

### ISSUE THREAD

**Issue by mars** [terraform-providers/terraform-provider-heroku#160]
### Terraform & Heroku Provider Version

```
Terraform v0.11.10
+ provider.heroku v1.7.1
```

### Affected Resource(s)
Please list the resources as a list, for example:
- heroku_build

### Terraform Configuration Files
```hcl
resource "heroku_app" "foobar" {
    name   = "foobar"
    region = "us"
}

resource "heroku_build" "foobar" {
  app = "${heroku_app.foobar.id}"

  source = {
    path = "./"
  }
}
```

### Debug Output

```
Error: Error running plan: 1 error(s) occurred:

* heroku_build.foobar_docker: 1 error(s) occurred:

* heroku_build.foobar_docker: Error generating build source tarball ./:
  Error generating build source tarball source-6939a70b-4a9f-f53a-95ae-08de1429ed1a.tar.gz of ./:
  archive/tar: write too long
```

### Expected Behavior

The Build is created successfully.

### Actual Behavior

The Build throws an error.

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

The `source.path` is the same directory containing the Terraform config.

**Comment by mars** [comment]
This may be due to writing the tarball into the current directory, which causes a recursively growing archive!

Possible fixes:

- prevent using current directory as source path by throwing an error with a clear message
- generate the tarball in the system tmp directory

**Comment by paulvollmer** [comment]
I have the same issue and I think the problem is that the  `.git` folder will be added to the tarball.

**Comment by mars** [comment]
Proposed a fix ➡ #178
